packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts

```diff
@@ -80,7 +80,9 @@ export const buildPrimitiveInstanceValue = (
       new GenericType(graph.getPrimitiveType(type)),
     ),
   );
-  instanceValue_setValues(instance, [value], observerContext);
+  if (value !== undefined) {
```

Comment:
Yeah `null` is the case the user hasn't set the value, so it should be `if (value !== null)`, not `if (value !== undefined)`. However, the current behavior sets it to an array like `[undefined]` if the value is `undefined`, not an empty array, so I think we should leave that behavior as is instead of setting it to an empty array if value is `undefined`.

Reply:
Removed the conditional statement so we aren't changing the behavior here. If `value` is null, we will create an array like `[null]` and if it's undefined, it will be `[undefined]`.